packages/tpetra/core/src/Tpetra_ConfigDefs.hpp

```diff
@@ -129,9 +129,19 @@ namespace Tpetra {
   /*! Allocation profile for matrix/graph entries */
   enum ProfileType {
     StaticProfile,  /*!< Single, static allocation (strict and more efficient) */
+#ifdef TPETRA_ENABLE_DEPRECATED_CODE
     DynamicProfile  /*!< Multiple, dynamic allocations (flexibile, but less efficient) */
```

Comment:
Don't we still want to mark the thing as deprecated, even if it's enabled?  Users have a way to disable deprecation warnings per package.  Furthermore, "enable deprecated code" implies that if it's _not_ enabled, then the code doesn't even exist.
```c++

enum ProfileType {
  StaticProfile
#ifdef TPETRA_ENABLE_DEPRECATED_CODE
  , DynamicProfile TPETRA_DEPRECATED
#endif 
};
```

Reply:
TPETRA_DEPRECATED doesn't work on a member of an enum, unfortunately.

Reply:
wonder if c++14 [deprecated] would work? Wish we could move trilinos to c++14...

Reply:
@GeoffDanielson wrote:
> `TPETRA_DEPRECATED` doesn't work on a member of an enum, unfortunately.

What does the compiler say?

Reply:
In file included from /home/gcdanie/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:0,
                 from /home/gcdanie/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:45,
                 from /home/gcdanie/Trilinos/packages/tpetra/core/src/Epetra_TsqrMessenger.hpp:60,
                 from /home/gcdanie/Trilinos/packages/tpetra/core/src/Epetra_TsqrMessenger.cpp:57:
/home/gcdanie/Trilinos/Obj_vesper/packages/tpetra/core/src/TpetraCore_config.h:188:32: error: expected ‘}’ before ‘__attribute__’
 #    define TPETRA_DEPRECATED  __attribute__((__deprecated__))
                                ^
/home/gcdanie/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:133:20: note: in expansion of macro ‘TPETRA_DEPRECATED’
     DynamicProfile TPETRA_DEPRECATED /*!< Multiple, dynamic allocations (flexibile, but less efficient) */

Reply:
Ugh, there's a way to do it, but it's grosssss.

```
  /*! Allocation profile for matrix/graph entries */
  enum ProfileType {
    StaticProfile
  };

#ifdef TPETRA_ENABLE_DEPRECATED_CODE
  TPETRA_DEPRECATED const ProfileType DynamicProfile = ProfileType(StaticProfile+1);
#define TPETRA_DEFAULT_PROFILE_TYPE DynamicProfile
#else
#define TPETRA_DEFAULT_PROFILE_TYPE StaticProfile
#endif
```

Reply:
@GeoffDanielson What compiler version is that?

Reply:
GCC 4.9.3:

```
 gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/4.9.3/base/libexec/gcc/x86_64-unknown-linux-gnu/4.9.3/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ./configure --prefix=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/4.9.3/base --disable-multilib
Thread model: posix
gcc version 4.9.3 (GCC) 
```